tasks/main.yml

```diff
@@ -18,28 +18,31 @@
   failed_when:
     - freshclam_result is failed
     - freshclam_result.stderr.find('locked by another process') != -1
-  tags: ['skip_ansible_lint']
+  tags: ["skip_ansible_lint"]
 
 - include_tasks: setup-RedHat.yml
   when: ansible_os_family == 'RedHat'
 
 - name: Change configuration for the ClamAV daemon.
   lineinfile:
     path: "{{ clamav_daemon_config_path }}"
-    regexp: '{{ item.regexp }}'
+    regexp: "{{ item.regexp }}"
     line: "{{ item.line | default('') }}"
     state: "{{ item.state | default('present') }}"
     mode: 0644
+    create: yes
   with_items: "{{ clamav_daemon_configuration_changes }}"
 
 - name: Ensure ClamAV daemon is running (if configured).
   service:
     name: "{{ clamav_daemon }}"
     state: "{{ clamav_daemon_state }}"
     enabled: "{{ clamav_daemon_enabled }}"
+  when: not ansible_check_mode
```

Comment:
These should instead be:

```yaml
    check_mode: yes
```

Since that effectively does the same thing, right?

Reply:
From my understanding, `check_mode: yes` would force the task to be run in check mode even if the `--check` flag is not provided, which is not what we want here.

The task fails in check mode because the service hasn't been created yet so we want to skip it to avoid a failure.
When we run this task in production, it succeeds because the service has been created in the task `Ensure ClamAV packages are installed.`

Reply:
Oh haha, you're right. I have no idea how my brain was confusing that.